### ISSUE TITLE
Flaky Kestrel Test: ClientCanReceiveFullConnectionCloseResponseWithoutErrorAtALowDataRate

### ISSUE THREAD

**Issue by anurse** [dotnet/aspnetcore#14671]
The Kestrel functional test: `ClientCanReceiveFullConnectionCloseResponseWithoutErrorAtALowDataRate` is only passing 66% of the time in `master` and less than 50% of the builds most days. We should deflake this test.

It's already quarantined but we need to actually fix it now.

The reported failure is:

```
System.IO.IOException : Unable to read data from the transport connection: Connection reset by peer.
---- System.Net.Sockets.SocketException : Connection reset by peer
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.ResponseTests.AssertStreamCompletedAtTargetRate(Stream stream, Int64 expectedBytes, Int32 targetBytesPerSecond) in /_/src/Servers/Kestrel/test/FunctionalTests/ResponseTests.cs:line 1039
   at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.ResponseTests.ClientCanReceiveFullConnectionCloseResponseWithoutErrorAtALowDataRate() in /_/src/Servers/Kestrel/test/FunctionalTests/ResponseTests.cs:line 970
--- End of stack trace from previous location where exception was thrown ---
----- Inner Stack Trace -----
```

Logs:

```
| [0.001s] TestLifetime Information: Starting test ClientCanReceiveFullConnectionCloseResponseWithoutErrorAtALowDataRate at 2019-09-27T18:00:17
| [0.050s] Microsoft.AspNetCore.Hosting.Diagnostics Debug: Hosting starting
| [0.051s] Microsoft.AspNetCore.Hosting.Diagnostics Debug: Hosting started
| [0.051s] Microsoft.AspNetCore.Hosting.Diagnostics Debug: Loaded hosting startup assembly Sockets.FunctionalTests, Version=42.42.42.42, Culture=neutral, PublicKeyToken=adb9793829ddae60
| [0.051s] Microsoft.AspNetCore.Server.Kestrel Debug: TestServer is listening on port 46003
| [0.056s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLQ3EI5B6SU5" accepted.
| [0.057s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLQ3EI5B6SU5" started.
| [0.058s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request starting HTTP/1.1 GET http:/// - -
| [11.133s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request finished HTTP/1.1 GET http:/// - - - 200 - - 11074.5875ms
| [11.134s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLQ3EI5B6SU5" disconnecting.
| [11.134s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLQ3EI5B6SU5" stopped.
| [11.135s] Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets Debug: Connection id "0HLQ3EI5B6SU5" sending FIN because: "The Socket transport's send loop completed gracefully."
| [14.138s] Microsoft.AspNetCore.Hosting.Diagnostics Debug: Hosting shutdown
| [16.006s] TestLifetime Information: Finished test ClientCanReceiveFullConnectionCloseResponseWithoutErrorAtALowDataRate in 16.0048934s
```

**Comment by halter73** [comment]
This test is already marked flaky. It was added to master post-3.0 to track a regression in the .NET Core 5.0 Linux socket behavior: https://github.com/aspnet/AspNetCore/pull/13532

This should be fixed once https://github.com/dotnet/corefx/pull/41250 is merged into corefx.

**Comment by BrennanConroy** [comment]
This has been unquarantined since Feb 28 and the PR that was blocking this was merged last year.
